pandas/_libs/tslibs/timedeltas.pyx

```diff
@@ -246,6 +246,47 @@ def array_to_timedelta64(object[:] values, unit='ns', errors='raise'):
     return iresult.base  # .base to access underlying np.ndarray
 
 
+def precision_from_unit(object unit):
```

Comment:
do this instead

make a new function that is cpdef that *calls* cast_from_unit, then you don't remove the inline nature from cast_from_unit and you can call it from python w/o re-writing.

Reply:
So the problem is that the current `cast_from_unit` does more, I only need the first part of it (that defines `m` and `p`, the rest of the function calculates something with `m` and `p` and returns that result).

I tried putting the common logic in a separate function, but it starts to become a bit more ugly to write it in such a way that it can still be used in the cdef inline one.

Reply:
I added a bit convoluted way to share the if/else part, but I don't see an easy other way to share that part, except by not having `cast_from_unit` a cdef function (but that has other implications elsewhere), or otherwise just duplicating the code

Reply:
you need to use cpdef.

Reply:
Can you cpdef inline?

Reply:
no. but inline for this prob doesn't make much difference. its the cast_from_unit where it actually matters a lot.

Reply:
How do do an `except? -1` for a cpdef that returns a tuple?

Reply:
actually. you are only calling this once? then make it a def is just fine (and error checkng is easy)

Reply:
In the new code, I am calling this only once. But the existing `cast_from_unit` is used in several places, and is also included in timedeltas.pxd as it used in other places in tslibs, so needs to stay a cdef

Reply:
Updated to use a cpdef as asked. It's only used in places where a similar cpdef `parse_timedelta_unit` is also used, so the python interaction is indeed probably not a problem.